online_party_system/party.go

```diff
@@ -27,21 +27,29 @@ const (
 )
 
 // Allow implementation-specific filtering of parameters received from clients in party join attempts.
-type MatchJoinMetadataFilter func (metadata map[string]string) map[string]string
+type MatchJoinMetadataFilter func(metadata map[string]string) map[string]string
+
+// MatchTerminateHook allows implementation-specific logic to run when a match terminates.
+type MatchTerminateHook func(matchID string, config PartyConfig, label *PartyMatchLabel)
+
+// MatchInitHook allows implementation-specific logic to run when a match is initialized.
+type MatchInitHook func(matchID string, config PartyConfig, label *PartyMatchLabel)
```

Comment:
This is fine to be called `MatchInitHook` because its actually executed in the `MatchInit` lifecycle of the match handler.